### PR TITLE
Preliminary Edge support

### PR DESCRIPTION
**Pull request by BenMcGarry**
Support for Microsoft Edge in latest builds.

### CODE REVIEW

**Comment by thybag** [comment]
TIL edge has finally added support for extensions (at least in the dev build)

Is this related? https://dev.windows.com/en-us/microsoft-edge/extensions/ (I notice they actually have a RES version to download)

**Comment by BenMcGarry** [comment]
Yep, they have build 4.5.5 out but i am working on getting 4.6.1 working. A few problems with local storage to fix.

File: chrome/background.js [hunk 1]
```diff
@@ -32,6 +32,14 @@
 
 /* global chrome: false */
 
+/* Edge Support - BETA */
+if (typeof msBrowser !== 'undefined') {
+	chrome = msBrowser;
```
Comment (erikdesjardins) [inline chrome/background.js]: For ESLint to pass, you'll need to change the comment up top to `/* global chrome: false, msBrowser: false */`, and add `// eslint-disable-line no-native-reassign` at the end of every line that you assign to `chrome`. Same with the other file.

Also some minor code style, see [the other build failures](https://travis-ci.org/honestbleeps/Reddit-Enhancement-Suite/jobs/117013599#L1046).

**Comment by BenMcGarry** [comment]
Due to Edge being preview, it is required to set special permissions onto the folder that the extension is in. This is done by running the following in command prompt from the extensions directory

`icacls "%cd%" /grant "*S-1-15-2-3624051433-2125758914-1423191267-1740899205-1073925389-3782572162-737981194":"(OI)(CI)(WDAC,WO,GE)"`

File: chrome/manifest.json [hunk 2]
```diff
@@ -4,12 +4,10 @@
 	"manifest_version": 2,
 	"minimum_chrome_version": "26.0",
 	"description": "Reddit Enhancement Suite - a group of enhancements for browsing Reddit",
-	"background": {
-		"scripts": [
-			"vendor/polyfill.min.js",
-			"background.js"
-		]
-	},
+    "background": {
```
Comment (erikdesjardins) [inline chrome/manifest.json]: Tab indentation pls. You can probably fix it from this page: click the pencil in the top right.

File: chrome/background.html [hunk 3]
```diff
@@ -0,0 +1,4 @@
+<html>
+    <script type="text/javascript" src="background.js"></script>
+    <script type="text/javascript" src="vendor/polyfill.min.js"></script>
```
Comment (erikdesjardins) [inline chrome/background.html]: This needs to be before background.js, so it has a chance to polyfill stuff before that runs

**Comment by andytuba** [comment]
Appears to work on v4.6.1: https://github.com/honestbleeps/Reddit-Enhancement-Suite/compare/v4.6.1...andytuba:v461-2797

**Comment by andytuba** [comment]
> Due to Edge being preview, it is required to set special permissions onto the folder that the extension is in. This is done by running the following in command prompt from the extensions directory

Did I hear you mention this isn't necessary with this week's build?

**Comment by BenMcGarry** [comment]
Still required on 14316

**Comment by andytuba** [comment]
Closed in favor of #2905

**Pull request closed without merging**